alibi/explainers/integrated_gradients.py

```diff
@@ -400,6 +400,14 @@ def _gradients_input(model: Union[tf.keras.models.Model],
 
     grads = tape.gradient(preds, x)
 
+    # if there are inputs have not impact to the output, the gradient is None, but we need to return a tensor
+    if isinstance(x, list):
+        shape = x[0].shape
+    else:
+        shape = x.shape
+    for idx, grad in enumerate(grads):
+        if grad is None:
+            grads[idx] = tf.convert_to_tensor(np.zeros(shape), dtype=x[idx].dtype)
```

Comment:
I think in an earlier commit you had `x[idx].shape`, which seems to make more sense in case each input has a different shape.